src/libcollections/btree/map.rs

```diff
@@ -276,6 +344,14 @@ pub struct OccupiedEntry<'a, K: 'a, V: 'a> {
 
 impl<K: Ord, V> BTreeMap<K, V> {
     /// Makes a new empty BTreeMap with a reasonable choice for B.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::collections::BTreeMap;
+    ///
+    /// let mut map: BTreeMap<&str, isize> = BTreeMap::new();
```

Comment:
I actually had the `insert` call the first time around, but after looking at some of the other collection docs, removed it for consistency. However, I personally like not having the type annotation as I feel it's a bit noisy and (as you rightly say) not so realistic. I vote for 2, and have fixed accordingly! I'll try to update the other collections as well at some point for consistency.